### ISSUE TITLE
Is there any way to have diff's work inside git clients, such as Sublime Merge?

### ISSUE THREAD

**Issue by othyn** [elasticdog/transcrypt#163]
I love this project, thank you for making it.

A gap in my knowledge on how this part of git works, but I'm wondering if I'm missing something and there is the ability to have diffs continue to function when inside git clients such as Sublime Merge?

As at the moment when viewing files in such git clients, the diff presented is that of the encrypted version of the file, instead of the decrypted one.

Not sure if this use case has come up yet!

**Comment by jmurty** [comment]
Yeah, diffs should just work in clients that apply the _.gitattributes_ directives. It's a shame your preferred client doesn't. 

You will need to find another client to diff and merge transcrypt-encrypted files. 

You should also be wary of committing secret files with that tool because it probably won't actually encrypt the contents. Transcrypt installs a pre-commit hook to try and prevent you committing and accidentally sharing files you thought would be encrypted but actually aren't, but I wouldn't recommend relying on it completely.

**Comment by othyn** [comment]
> Yeah, diffs should just work in clients that apply the .gitattributes directives. It's a shame your preferred client doesn't.

> You will need to find another client to diff and merge transcrypt-encrypted files.

Appears to be the way! A shame as its a good client otherwise.

> You should also be wary of committing secret files with that tool because it probably won't actually encrypt the contents. Transcrypt installs a pre-commit hook to try and prevent you committing and accidentally sharing files you thought would be encrypted but actually aren't, but I wouldn't recommend relying on it completely.

It does respect the pre-commit hooks and stops exec if transcrypt fails or throws an error such as that, I've tested to make sure on that one! So we're all good there.
